### ISSUE TITLE
Synology reverse proxy needed to address SSL certificate issues running Node-RED in a Docker container on Synology NAS

### ISSUE THREAD

**Issue by HonestJohn61** [mikejac/node-red-contrib-google-smarthome#361]
My knowledge is very limited, so there may be something obvious I'm missing, but after following the installation instructions I'm getting this error in the Node-RED debug window:

_**"Error: ENOENT: no such file or directory, open 'ffd-ggl-svr-privkey.pem'"**_

Unsurprisingly, http://************.synology.me:6336/check refuses to connect.

I got the certificate from Let's Encrypt using the Synology server's built-in Control Panel | Security | Certificate | Add process, which I then exported, placing chain.pem and privkey.pem in the Node-RED's user folder (prepending the names with "ffd-ggl-svr-"). So my Configuration node Webserver settings look like this:

<img width="490" alt="image" src="https://user-images.githubusercontent.com/29650722/210753697-90ff2446-fc80-496d-89f9-1caf8bc22406.png">

Thanks

**Comment by HonestJohn61** [comment]
Thanks @Caprico85 for the reply. I've tried that but still no joy with your suggestion.

I've read your guide to using a proxy server since the Synology NAS offers that facility. "Use external SSL offload" seems to overcome the SSL issue (see if you agree, based on what I've noted below), but getting the reverse proxy to reach the node-red-contrib-google-smarthome server now seems to be the issue. So here are my revised settings for the Synology reverse proxy:

<img width="590" alt="image" src="https://user-images.githubusercontent.com/29650722/210898897-1632c3a2-6c17-4ac3-8070-cadefc982406.png">

And here are the Webserver settings:

<img width="466" alt="image" src="https://user-images.githubusercontent.com/29650722/210899452-7ade05a2-e216-4185-b66e-34a79848eec9.png">

Now Node-RED's debug window reports that the server has started ...

<img width="295" alt="image" src="https://user-images.githubusercontent.com/29650722/210900834-eaacdb21-e9c2-41c7-aa33-e90a056bc631.png">

... but http://**_hostIP_**:6337 still refuses to connect. (The hostIP is the same for Node_RED and the Synology NAS, since Node-RED is running in a docker container on the Synology NAS). So, again unsurprisingly, https://*********.synology.me:6336/check times out, eventually giving this response:

<img width="749" alt="image" src="https://user-images.githubusercontent.com/29650722/210902067-a8b8e72e-d31b-4ee5-8ec9-cdda200c612d.png">

All suggestions welcome! Thanks

**Comment by HonestJohn61** [comment]
Adding this port to the Docker container for Node_RED ...

<img width="497" alt="image" src="https://user-images.githubusercontent.com/29650722/210904333-27b575ff-e30e-4b6b-a28b-26ab4aba83b7.png">

... allows a connection of some sort to **_hostIP_**:6337:

<img width="475" alt="image" src="https://user-images.githubusercontent.com/29650722/210904861-11525c67-6d5a-4c83-859c-9fc461a73182.png">

**Comment by HonestJohn61** [comment]
So, logically, the webserver has to operate using http not https. And I need to eliminate the spurious reference to example.com. And it worked! Well, at least I can access the Google SmartHome test page on https://**********.synology.me:6336/check. Yippee! Here are my settings:

<img width="496" alt="image" src="https://user-images.githubusercontent.com/29650722/210906122-64345cce-c9c5-43da-a0a1-3b002f4b08f2.png">

<img width="462" alt="image" src="https://user-images.githubusercontent.com/29650722/210906019-d14cb6bf-1086-4255-a590-3f7dc14d7e46.png">

**Comment by HonestJohn61** [comment]
But not quite there yet. The test yields the following:

<img width="1437" alt="image" src="https://user-images.githubusercontent.com/29650722/210908249-fccd09ae-30c6-429d-bd92-3507de48d3ef.png">

However, Google Home won't connect, and the Node-RED debug window suggests there's another issue to address:

<img width="299" alt="image" src="https://user-images.githubusercontent.com/29650722/210908527-3c6245dd-a77e-4ffd-b0f5-37361ef02c08.png">

So that's where I'm stuck now, and it's time for some sleep! Any suggestions?

**Comment by HonestJohn61** [comment]
Sorted! There must have also been an error in the setup at the Google end. I deleted the original project and created a new one, which now works.

I'd just note that the documentation could have been more helpful if it was a little more expansive about reverse proxy and why you might need to do it. I also found the explanation of the Path field and Use http Node-RED root path checkbox confusing - I still don't understand why example.com seems to be almost mandated, with the Path field just adding to rather than overriding the default, and only the checkbox facilitating override (if I've understood this correctly).
